docs/guide_restli_server.md

```diff
@@ -1295,6 +1288,59 @@ FortunesBuilders().partialUpdate().id(1L).input(patch).build();
 `PatchGenerator.diff(original, revised)` can also be used to create a
 minimal partial update.
 
+#### Returning entity in PARTIAL_UPDATE response
+
+By default, the patched entity is not returned in the PARTIAL_UPDATE response because
+the client already has the patch data and possibly has the rest of the entity as well.
+However, there are use cases where the server will attach additional data to the new entity or the user
+simply doesn't have the whole entity. Returning the entity in the PARTIAL_UPDATE response saves  the client
+another GET request.
+
+Starting in Rest.li version 24.0.0, we provide the developer the option to
+return the newly created entity. To use this feature, add a @`ReturnEntity`
+annotation to the method that implements PARTIAL_UPDATE. The return type of the
+method must be `UpdateEntityResponse`.
+
+```java
+@ReturnEntity
+@RestMethod.PartialUpdate
+public UpdateEntityResponse<V> partialUpdate(K key, PatchRequest<V> patch);
+```
+An example resource method implementation is as follows, note that the return type will be  ````UpdateEntityResponse```` :
+
+```java
+@ReturnEntity
+@RestMethod.PartialUpdate
+public UpdateEntityResponse<Greeting> update(Long key, PatchRequest<Greeting> patch)
+{
+    Greeting greeting = _db.get(key);
+
+    if (greeting == null)
+    {
+        throw new RestLiServiceException(HttpStatus.S_404_NOT_FOUND);
+    }
+
+    try
+    {
+        PatchApplier.applyPatch(greeting, patch);
+    }
+    catch (DataProcessingException e)
+    {
+        throw new RestLiServiceException(HttpStatus.S_400_BAD_REQUEST);
+    }
+
+    return new UpdateEntityResponse<Greeting>(HttpStatus.S_200_OK, greeting);
+}
+```
+
+By default, all requests to a PARTIAL_UPDATE resource method annotated with @`ReturnEntity` will return the patched entity in the response. However, if the client decides that it doesn't want the entity to be returned (to reduce network traffic, for instance), then the query parameter `$returnEntity` can be used to indicate this. A value of `true` indicates that the entity should be returned, a value of `false` indicates that the entity shouldn't be returned, and omitting the query parameter altogether defaults to treating the value as if it were `true`. Note that if the resource method isn't annotated with @`ReturnEntity`, then the value of `$returnEntity` will be ignored.
```

Comment:
This section is common for create/batch_create/partial_update/batch_partial_update. 
It might be interesting to keep the example here, but to have a common section to explain the behavior.
We can also point out 2 optimizations
 - This query parameter is managed by the framework and will strip out the entity accordingly. It will be transparent for the SO
 - Based on this query param, The SO could optimize his code

Reply:
I think the common section I mentioned can be described in a different doc we can host under `API References`